addons/l10n_ec/models/res_partner.py

```diff
@@ -69,22 +69,26 @@ def check_vat(self):
                     if partner.l10n_latam_identification_type_id.id == it_ruc.id and len(partner.vat) != 13:
                         raise ValidationError(_('If your identification type is %s, it must be 13 digits')
                                               % it_ruc.display_name)
-                    final_consumer = verify_final_consumer(partner.vat)
-                    if final_consumer:
-                        valid = True
-                    else:
-                        valid = self.is_valid_ruc_ec(partner.vat)
-                    if not valid:
-                        error_message = ""
-                        if partner.l10n_latam_identification_type_id.id == it_dni.id:
-                            error_message = _("VAT %s is not valid for an Ecuadorian DNI, "
-                                              "it must be like this form 1234567897") % partner.vat
-                        if partner.l10n_latam_identification_type_id.id == it_ruc.id:
-                            error_message = _("VAT %s is not valid for an Ecuadorian company, "
-                                              "it must be like this form 1234567897001") % partner.vat
-                        raise ValidationError(error_message)
         return super(ResPartner, self - ecuadorian_partners).check_vat()
 
+    @api.depends("vat", "country_id", "l10n_latam_identification_type_id")
+    def _compute_l10n_ec_vat_validation(self):
+        it_ruc = self.env.ref("l10n_ec.ec_ruc", False)
+        it_dni = self.env.ref("l10n_ec.ec_dni", False)
+        ruc = stdnum.util.get_cc_module("ec", "ruc")
+        ci = stdnum.util.get_cc_module("ec", "ci")
+        for partner in self:
+            partner.l10n_ec_vat_validation = False
+            if partner and partner.l10n_latam_identification_type_id in (it_ruc, it_dni) and partner.vat:
+                final_consumer = verify_final_consumer(partner.vat)
```

Comment:
The identification type of CONSUMIDOR FINAL should be RUC, but the number 9999999999999 needs to have an exception because its teenth digit doesn't comply with the checksum validation